### PR TITLE
Maintain image aspect ratio

### PR DESCRIPTION
**Pull request by btjones**
## tldr;
This PR updates the image logic when only a new width or new height is provided to maintain the original image aspect ratio.

## Details
Previously, if only one dimension was provided, the non-specified dimension would fall back to the original image size. For example a 640x320 image that was rendered with only a height of `32` specified would end up with a width of `640` and a height of `32`:

```
render.Image(
    src = SOME_IMAGE,
    height = 32,
)
```

With this change the new image width would be scaled to match the new height while maintaining the original image aspect ratio. So in this example the image would be rendered with a width of `64` and a height of `32`.

## Why do this?
The image size is not always known ahead of time so this makes it simpler to resize an image on the fly to properly fit in the 64x32 pixlet/tidbyt screen.

## Example

Original Image (323 × 200):
original-image

Previous functionality when defining only a height of `32` would output a stretched image:
image-test-old

With this change the image maintains the original aspect ratio:
image-test-new

### CODE REVIEW

**Comment by btjones** [comment]
Hey @rohansingh / @betterengineering apologies for the ping but since this is my first PR to this repo, I'm not sure of the process. Looks like I need someone to approve the workflows.

**Comment by matslina** [comment]
Hey @btjones, and thanks for the contribution!
Very reasonable change, and we'd be happy to merge it. Just that one question mark around division by zero.

File: render/image.go [hunk 1]
```diff
@@ -127,10 +127,12 @@ func (p *Image) Init() error {
 	if p.Width != 0 || p.Height != 0 {
 		nw, nh := p.Width, p.Height
 		if nw == 0 {
-			nw = w
+			// scale width, maintaining original aspect ratio
+			nw = int(float64(nh)*(float64(w)/float64(h)))
```
[the remark this answers is not in the document]
Comment (matslina) [reply render/image.go:131]: Actually I'll just merge this and fix it later.

**Comment by matslina** [comment]
Changed my mind and merged. =)
Thank you @btjones!
